src/libraries/System.CodeDom/tests/System/CodeDom/Compiler/IndentedTextWriterTests.cs

```diff
@@ -0,0 +1,606 @@
+        [MemberData(nameof(WriteAsync_MemberData))]
+        public async Task WriteAsync_WithIndents_OmitsTabsAfterWriteAsync(Func<IndentedTextWriter, Task> callWriteAsync, string expected)
+        {
+            var indicator = new IndicatingTextWriter();
+            var itw = new IndentedTextWriter(indicator, TabString);
+            itw.NewLine = NewLine;
+            itw.Indent = 1;
+
+            string prefix = "prefix";
+            await itw.WriteAsync(prefix);
+            await callWriteAsync(itw);
+
+            Assert.Equal(nameof(IndentedTextWriter.WriteAsync), indicator.LastCalledMethod);
+            Assert.Equal($"{prefix}{expected}", indicator.GetStringBuilder().ToString());
+        }
+
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        [Theory]
+        [MemberData(nameof(WriteLineAsync_MemberData))]
+        public async Task WriteLineAsync_WithoutIndents_CallsInnerWriteLineAsync(Func<IndentedTextWriter, Task> callWriteLineAsync, string expected)
+        {
+            var indicator = new IndicatingTextWriter();
+            var itw = new IndentedTextWriter(indicator, TabString);
+            itw.NewLine = NewLine;
+            await callWriteLineAsync(itw);
+
+            Assert.Equal(nameof(TextWriter.WriteLineAsync), indicator.LastCalledMethod);
+            Assert.Equal(expected, indicator.GetStringBuilder().ToString());
+        }
+
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        [Theory]
+        [MemberData(nameof(WriteLineAsync_MemberData))]
+        public async Task WriteLineAsync_WithIndents_FirstLine_IsNotIndented(Func<IndentedTextWriter, Task> callWriteLineAsync, string expected)
+        {
+            var indicator = new IndicatingTextWriter();
+            var itw = new IndentedTextWriter(indicator, TabString);
+            itw.NewLine = NewLine;
+            itw.Indent = 1;
+            await callWriteLineAsync(itw);
+
+            Assert.Equal(nameof(IndentedTextWriter.WriteLineAsync), indicator.LastCalledMethod);
+            Assert.Equal($"{expected}", indicator.GetStringBuilder().ToString());
+        }
+
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        [Theory]
+        [MemberData(nameof(WriteLineAsync_MemberData))]
+        public async Task WriteLineAsync_WithIndents_SubsequentLines_AreIndented(Func<IndentedTextWriter, Task> callWriteLineAsync, string expected)
+        {
+            var indicator = new IndicatingTextWriter();
+            var itw = new IndentedTextWriter(indicator, TabString);
+            itw.NewLine = NewLine;
+            itw.Indent = 1;
+
+            const string prefix = "prefix";
+            await itw.WriteLineAsync(prefix);
+            await callWriteLineAsync(itw);
+
+            Assert.Equal(nameof(IndentedTextWriter.WriteLineAsync), indicator.LastCalledMethod);
+            Assert.Equal($"{prefix}{NewLine}{TabString}{expected}", indicator.GetStringBuilder().ToString());
```

Comment:
Could we add a test testing the original scenario, that shows that indentation is added after calling `IndentedTextWriter.WriteLineAsync();`, something like:
```csharp
        [Fact]
        public async Task TestIdentationAfterParameterlessWriteLineAsync()
        {
            var asyncSw = new StringWriter();
            var asyncTw = new IndentedTextWriter(asyncSw);

            asyncTw.Indent++;
            await asyncTw.WriteLineAsync("Foo");
            await asyncTw.WriteLineAsync("Bar");
            await asyncTw.WriteLineAsync();
            await asyncTw.WriteLineAsync("Quux");
            await asyncTw.WriteLineAsync("Wibble");

            var sw = new StringWriter();
            var tw = new IndentedTextWriter(sw);

            tw.Indent++;
            tw.WriteLine("Foo");
            tw.WriteLine("Bar");
            tw.WriteLine();
            tw.WriteLine("Quux");
            tw.WriteLine("Wibble");

            Assert.Equal(sw.GetStringBuilder().ToString(), asyncSw.GetStringBuilder().ToString());
        }
```